homeassistant/components/geniushub/config_flow.py

```diff
@@ -0,0 +1,205 @@
+"""Config flow for Geniushub logger integration."""
+
+from __future__ import annotations
+
+from http import HTTPStatus
+import logging
+import socket
+from typing import Any
+
+import aiohttp
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.config_entries import ConfigFlowResult
+from homeassistant.const import (
+    CONF_HOST,
+    CONF_MAC,
+    CONF_PASSWORD,
+    CONF_TOKEN,
+    CONF_USERNAME,
+)
+from homeassistant.helpers import config_validation as cv
+
+from . import validate_input
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+Cloud_API_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_TOKEN): cv.string,
+        vol.Required(CONF_MAC): cv.string,  # vol.Match(MAC_ADDRESS_REGEXP),
+    }
+)
+
+
+Local_API_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_HOST): cv.string,
+        vol.Required(CONF_USERNAME): cv.string,
+        vol.Required(CONF_PASSWORD): cv.string,
+        vol.Optional(CONF_MAC): cv.string,  # vol.Match(MAC_ADDRESS_REGEXP),
+    }
+)
+
+
+class GeniusHubConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Solis logger."""
+
+    VERSION = 1
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """User config step for determine v1 or v3."""
+        _LOGGER.debug("ConfigFlow.async_step_user")
+        return self.async_show_menu(
+            step_id="user",
+            menu_options=["local_api", "cloud_api"],
+        )
+
+    async def async_step_local_api(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Version 3 configuration."""
+        _LOGGER.debug("ConfigFlow.async_step_local_api: ", extra=user_input)
+        if user_input is None:
+            return self.async_show_form(
+                step_id="local_api", data_schema=Local_API_SCHEMA
+            )
+
+        self._async_abort_entries_match(user_input)
+        errors = {}
+
+        try:
+            info = await validate_input(self.hass, user_input)
```

Comment:
Do we get something unique from this?

Reply:
No - only the MAC address that the user has entered.